design/ceph/object/ceph-cosi-driver.md

```diff
@@ -0,0 +1,150 @@
+
+#### cephcosidriver.ceph.rook.io CRD
+
+The users can define following CRD so that configuration related the Ceph COSI driver can be passed. This is not mandatory and the driver will be started with default settings if this CRD is not defined.
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephCosiDriver
+metadata:
+  name: rook-ceph-cosi
+  namespace: rook-ceph
+spec:
+  deploymentStrategy: "Auto"
+```
+
+The `deploymentStrategy` can be `Auto` or `Force` or `Disable`. If `Auto` is specified the driver will be started automatically when first object store is created and will be stopped when last object store is deleted. If `Force` is specified the driver will be started when the CRD is created and will be stopped when the CRD is deleted. The user can also disable the driver by specifying `Disable` as the value.
+
+#### Creating COSI Related CRDs
+
+There are five different kubernetes resources related to COSI. They are `Bucket`, `BucketAccess`, `BucketAccessClass`, `BucketClass` and `BucketClaim`. The user can create these resources using following `kubectl` command. All the examples will be added to deploy/examples/cosi directory in the Rook repository.
+
+```bash
+kubectl create -f deploy/examples/cosi/bucketclass.yaml
+kubectl create -f deploy/examples/cosi/bucketclaim.yaml
+kubectl create -f deploy/examples/cosi/bucketaccessclass.yaml
+kubectl create -f deploy/examples/cosi/bucketaccess.yaml
+```
+
+User need refer Secret which contains access credentials for the object store in the `Parameter`  field of `BucketClass` and `BucketAccessClass` CRD as below:
+
+```yaml
+Parameter
+  credentialsReference: "rook-ceph-object-user-<objectstore-name>-cosi"
+  namespaceReference: "<objectstore-store-namespace>"
+```
+
+#### Consuming COSI Bucket
+
+The user needs to mount the secret as volume created by `BucketAccess` to the application pod. The user can use the secret to access the bucket by parsing the mounted file.
+
+```yaml
+spec:
+  containers:
+      volumeMounts:
+        - name: cosi-secrets
+          mountPath: /data/cosi
+  volumes:
+  - name: cosi-secrets
+    secret:
+      secretName: ba-secret
+```
+
+```bash
+# cat /data/cosi/bucket_info.json
+```
+
+```json
+    {
```

Comment:
nit: The json highlighting is strange. I wonder if it will be fixed if you reduce the indentation a level.
```suggestion
{
```